src/test/groovy/com/synacy/gradprogram/spock/exercise/ShoppingServiceSpec.groovy

```diff
@@ -0,0 +1,46 @@
+package com.synacy.gradprogram.spock.exercise
+
+import com.synacy.gradprogram.spock.demo.User
+import org.apache.tools.ant.taskdefs.condition.Or
+import spock.lang.Specification
+
+
+class ShoppingServiceSpec extends Specification {
+
+    ShoppingService shoppingService
+    OrderingService orderingService = Mock(OrderingService)
+    DeliveryService deliveryService = Mock(DeliveryService)
+    OrderRepository orderRepository = Mock(OrderRepository)
+
+    DeliveryRequestRepository deliveryRequestRepository = Mock(DeliveryRequestRepository)
+
+    void setup() {
+        shoppingService = new ShoppingService(orderingService, deliveryService, orderRepository, deliveryRequestRepository)
+    }
+
+    def "BuyNonSpoilingItemsInCart should create an Order Delivery"() {
+        given:
+        Cart cart =Mock(Cart)
+        Order order = Mock(Order)
+        User user = new User(firstName: "Ernest", lastName: "Pogi")
+
+
+        when:
+        shoppingService.buyNonSpoilingItemsInCart(cart,user)
+
+        then:
+
+        1 * orderingService.applyDiscountToCartItems(cart)
+        1 * orderingService.createAnOrder(_,_,_,_) >> new Order()
+        1 * deliveryService.createDelivery(_ as Order)
```

Comment:
We should only be testing one thing at a time. We can separate these checks into individual unit tests. Also only use wildcard as a last resort.